File: python/fib.py
import sys

def fib(value, a=0, b=1):
    o = a
    a = b
    b = a + o
    if(a <= value):
        print(a)
        fib(value, a = a, b = b)
   
if __name__ == "__main__":
    fib(int(sys.argv[1]))

File: c/fib.c
#include <stdio.h>
#include <stdlib.h>

int a = 0, b = 1, o = 0, max = 10;


void fib(int a, int b){
  o = a;
  a = b;
  b = a + o;
  if(a <= max){
    printf("%d\n", a);
    fib(a, b);
  }
}

void main(int argc, char *argv[]){
  max = atoi(argv[1]);
  printf("%d\n", max);
  //fib(a, b);
}
